### PR TITLE
feat(#546): POC with jcabi.xml in XmlNode#optchild(name)

### PR DESCRIPTION
**Pull request by h1alexbel**
In this pull I've introduced small refactoring of `XmlNode#optchild(name)` method. Now method uses `jcabi.xml` as XML API, instead of `w3c.dom`.

ref #546
History:
- **feat(#546): retrieves child test**
- **feat(#546): opt child POC**
- **feat(#546): objects POC**
- **feat(#546): objects POC off**
- **feat(#546): clean for qulice**
- **feat(#546): redundant doc**

---

## PR-Codex overview
This PR focuses on enhancing the `XmlNode` class in the `org.eolang.jeo.representation.xmir` package by replacing the `Node` type with `XML`, improving child retrieval methods, and adding unit tests for XML node functionalities.

### Detailed summary
- Added unit tests in `XmlNodeTest` for retrieving child nodes and child objects.
- Changed `private final Node node;` to `private final XML node;` in `XmlNode`.
- Updated constructors in `XmlNode` to accept `XML` type.
- Revised methods to utilize `XML` functionalities instead of `Node`.
- Simplified equality and string representation methods in `XmlNode`.
- Modified child retrieval logic to use `XML` methods.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by h1alexbel** [comment]
@volodya-lombrozo take a look, please

File: src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java [hunk 1]
```diff
@@ -93,6 +95,42 @@ void retrievesText() {
...
 
+    @Test
+    void retrievesChild() {
+        final XmlNode child = new XmlNode(
+            "<program><o>text</o></program>"
+        ).child("o");
+        final String expected = "<o>text</o>";
+        MatcherAssert.assertThat(
+            String.format(
+            "Retrieved XML: %s does not match with expected %s",
```
Comment (volodya-lombrozo) [inline src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java:106]: @h1alexbel Something strange is happening with indentation here? Maybe we need two more spaces here?

File: src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java [hunk 2]
```diff
@@ -93,6 +95,42 @@ void retrievesText() {
...
+        final List<XmlNode> objects = new XmlNode(
+            "<program><o>o1</o><o>o2</o></program>"
+        ).children().collect(Collectors.toList());
+        final List<XmlNode> expected = new ListOf<>(
+            new XmlNode("<o>o1</o>"),
+            new XmlNode("<o>o2</o>")
+        );
+        MatcherAssert.assertThat(
+            String.format(
+            "Retrieved child objects: %s don't match with expected: %s",
```
Comment (volodya-lombrozo) [inline src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java:126]: @h1alexbel The same problem with indentation.

File: src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java [hunk 3]
```diff
@@ -93,6 +95,42 @@ void retrievesText() {
...
+                objects,
+                expected
+            ),
+            objects, new IsEqual<>(expected)
```
Comment (volodya-lombrozo) [inline src/test/java/org/eolang/jeo/representation/xmir/XmlNodeTest.java:130]: @h1alexbel Maybe it's better to place `new IsEqual<>(expected)` on the next line for consistency with the previous test? What do you think?

File: src/main/java/org/eolang/jeo/representation/xmir/XmlNode.java [hunk 4]
```diff
@@ -224,13 +233,9 @@ XmlBytecodeEntry toEntry() {
...
     private Optional<XmlNode> optchild(final String name) {
         Optional<XmlNode> result = Optional.empty();
-        final NodeList children = this.node.getChildNodes();
-        for (int index = 0; index < children.getLength(); ++index) {
-            final Node current = children.item(index);
-            if (current.getNodeName().equals(name)) {
-                result = Optional.of(new XmlNode(current));
-                break;
-            }
+        final List<XML> nodes = new XMLDocument(this.node).nodes(name);
```
Comment (volodya-lombrozo) [inline src/main/java/org/eolang/jeo/representation/xmir/XmlNode.java:236]: @h1alexbel The code is much simpler, indeed! What is a bit strange, that we parse `xml` node in the constructor:
```java
public XmlNode(final XML xml) {
  this(xml.node());
}
```

and then create it again:

```java
final List<XML> nodes = new XMLDocument(this.node).nodes(name);
```

I understand, why you do this, but maybe we can replace `private final Node node;` field with `private final XML node;` ? What do you think?If you’re okay with it, it’s better to add one more puzzle.

Comment (h1alexbel) [reply src/main/java/org/eolang/jeo/representation/xmir/XmlNode.java:236]: @volodya-lombrozo I thought about the replacement of `Node` with `XML`. I will replace it in this pull

**Comment by h1alexbel** [comment]
@volodya-lombrozo updated pull request. Take a look again, please

**Review by volodya-lombrozo — APPROVED** [review]
@h1alexbel Looks good to me, thank you! I think we can continue to remove `org.w3c.dom.Node` usage from other places.

**Pull request merged**